packages/effection/tests/execution.test.js

```diff
@@ -76,40 +76,4 @@ describe('execution', () => {
       });
     });
   });
-
-  describe('forking from a process', () => {
-    let exec, resume, fail;
-
-    beforeEach(() => {
-      exec = main(fork(context => ({ resume, fail } = context)));
-    });
-
-    it('makes the external process waiting and blocking', () => {
-      expect(exec.isBlocking).toBe(true);
-      expect(exec.isWaiting).toBe(true);
-    });
-
-    describe('resuming the forked process', () => {
-      beforeEach(() => {
-        resume();
-      });
-      it('completes the external process', () => {
-        expect(exec.isCompleted).toBe(true);
-      });
-    });
-
-    describe('failing the forked process', () => {
-      beforeEach(() => {
-        fail(new Error('boom!'));
-      });
-
-      it('fails the process', () => {
-        expect(exec.isErrored).toBe(true);
-      });
-      it('has the error as the result', () => {
-        expect(exec.result).toBeDefined();
-        expect(exec.result.message).toEqual('boom!');
-      });
-    });
-  });
 });
```

Comment:
For some reason figuring out how to make this work really hurt my brain. But then I realized that maybe this just shouldn't be allowed as a pattern? Ideally it would just work, but it turned out to be surprisingly tricky.